woocommerce/woocommerce
# Add woocommerce_attributes_saved trigger. Closes #35004.

### All Submissions:






### Changes proposed in this Pull Request:

Introduce `woocommerce_attributes_saved` trigger in admin metabox. Explanation in #35004.

Closes #35004.



### How to test the changes in this Pull Request:

Setup:

- After checking out this branch, start by recompiling legacy assets: `pnpm run --filter='woocommerce/client/legacy' build`.
- When you open the product editor, be sure to clear your browser's cache so that the new JS is picked up.

This change results in a new browser event being triggered: this is for the benefit of 3PDs and nothing in WooCommerce Core takes advantage of it at present, therefore there isn't really anything tangible to test. So, our first goal should simply be to confirm we have not accidentally broken anything in the product editor:

- Check you can update and then save changes to products.
- Specifically, ensure you can modify and save changes to product attributes.

![product-editor-metabox-attributes](https://user-images.githubusercontent.com/3594411/221214156-751cce9f-d410-4cee-bcd0-84f2765ceb23.png)

Optionally, we can also perform some further tests to make sure the new event is available to 3PDs. Start by adding the following code (I suggest to a location such as `wp-content/mu-plugins/test-pr-36943.php` which you can remove after testing):

```php
<?php

add_action( 'admin_footer', function () {
	echo "
		<script>
			jQuery( document.body ).on( 'woocommerce_attributes_saved', function() {
				alert( 'New event \"woocommerce_attributes_saved\" was triggered.' );
			} );
		</script>
	";
} );
```

With that in place:

- Refresh the product editor.
- Modify some product attributes, then click on the `Save Attributes` button.
- You should see an alert, as in the following screenshot.

<img width="998" alt="alert" src="https://user-images.githubusercontent.com/3594411/221215495-e7046e4d-7805-434b-9843-6749712e49b2.png">



### Other information:




### FOR PR REVIEWER ONLY:
